src/main/java/mate/academy/myJdbc/dao/DeveloperDaoImpl.java

```diff
@@ -0,0 +1,240 @@
+                        rs.getString("skill_branch"),
+                        rs.getString("skill_level"));
+                developer.addSkill(skill);
+            }
+
+            //add projects to developer
+            statement = connection.prepareStatement("SELECT projects.project_id, projects.project_name, projects.project_cost, projects.project_birthday, developers.developer_id\n" +
+                    "FROM projects\n" +
+                    "JOIN projects_developers ON (projects.project_id = projects_developers.project_id)\n" +
+                    "JOIN developers ON (developers.developer_id = projects_developers.developer_id)\n" +
+                    "HAVING developer_id = ?;");
+            statement.setInt(1, id);
+            rs = statement.executeQuery();
+            while (rs.next()) {
+                Project project = new Project.ProjectBuilder().setId(rs.getInt("project_id")).setBirthday(rs.getDate("project_birthday")).setName(rs.getString("project_name")).build();
+                developer.addProject(project);
+            }
+
+            return developer;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public void updateDeveloper(Developer developer) {
+        try {
+            PreparedStatement statement = connection.prepareStatement("UPDATE developers SET developer_name=?" +
+                    ", developer_age=?, developer_salary=? WHERE developer_id=?;");
+            statement.setString(1, developer.getName());
+            statement.setInt(2, developer.getAge());
+            statement.setInt(3, developer.getSalary());
+            statement.setInt(4, developer.getId());
+            statement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void deleteDeveloper(int id) {
+        try {
+            deleteDeveloperSkillRelations(id);
+            deleteDeveloperProjectRelations(id);
+
+            PreparedStatement statement = connection.prepareStatement("DELETE FROM developers WHERE developer_id=?;");
+            statement.setInt(1, id);
+            statement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public int getSkillId(Skill skill) {
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT skill_id FROM skills WHERE skill_branch=? " +
+                    "AND skill_level=?;");
+            statement.setString(1, skill.getBranch());
+            statement.setString(2, skill.getLevel());
+            ResultSet rs = statement.executeQuery();
+            rs.next();
+            int skillId = rs.getInt("skill_id");
+
+            return skillId;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return 0;
+    }
+
+    public void insertDevelopersSkillsRelation(int devId, int skillId) {
```

Comment:
I would recommend you to rename thi method to addSkillForDeveloper(int devId, int skillId)